docs/elasticsearch_statefulset.md

```diff
@@ -0,0 +1,215 @@
+
+    def populateData(connection):
+        connection.index(index='messages', doc_type='song', id=1, body={
+            'author': 'Aladdin',
+            'song': 'A Whole New World',
+            'lyrics': ['I can show you the world'],
+            'awesomeness': 42
+        })
+
+
+    if __name__ == "__main__":
+        populateData(es_conn)
+
+In [run.py](../app/run.py) we add another resource which will get the message from elasticsearch. 
+
+    class ElasticsearchResource(object):
+        def on_get(self, req, resp):
+            resp.status = falcon.HTTP_200
+            data = es_conn.get(index='messages', doc_type='song', id=1)
+            resp.body = json.dumps(data['_source'])
+
+    if es_conn:
+      app.add_route('/app/elasticsearch', ElasticsearchResource())
+ 
+### Application initContainers
+Finally, we want to make sure that our app doesn't start before the elasticsearch service is ready. We add two initContainers to check and populate elasticsearch in [\_elasticsearch_init.tpl.yaml](../helm/aladdin-demo/templates/_elasticsearch_init.tpl.yaml).
+
+    {{ define "elasticsearch_check" -}}
+    {{ if .Values.elasticsearch.create }}
+    - name: {{ .Chart.Name }}-elasticsearch-check
+      image: byrnedo/alpine-curl
+      command:
+      - 'sh'
+      - '-c'
+      - 'until curl {{ .Chart.Name }}-elasticsearch:{{ .Values.elasticsearch.port }}; do echo waiting for elasticsearch pod; sleep 2; done;'
+    {{ end }}
+    {{ end }}
+
+    ---
+
+    {{ define "elasticsearch_populate" -}}
+    {{ if .Values.elasticsearch.create }}
+    - name: {{ .Chart.Name }}-elasticsearch-populate
+      image: {{ .Values.deploy.ecr }}{{ .Chart.Name }}:{{ .Values.deploy.imageTag }}
+      command:
+      - 'python3'
+      - 'elasticsearch_util/elasticsearch_populate.py'
+      envFrom:
+        - configMapRef:
+            name: {{ .Chart.Name }}
+    {{ end }}
+    {{ end }}
+
+We include these initContainers in [aladdin-demo.deploy.yaml](../helm/aladdin-demo/templates/aladdin-demo.deploy.yaml). Since we are using persistent volume for data storage, we should only need to populate the data once, so the population initContainer is controlled by a `elasticsearch.populate` boolean specified in [values.yaml](../heml/aladdin-demo/values.yaml).
+
+## Test Persistency
+Make sure that `elasticsearch.create` and `elasticsearch.populate` are set to `true` in [values.yaml](../heml/aladdin-demo/values.yaml). Build and deploy aladdin-demo and wait until all the pods are running.
+
+    $ aladdin build
+    $ aladdin start
+    $ kubectl get pods
+
+Check that elasticsearch has been populated by curling the appropriate endpoint of the aladdin-demo service.
+
+    $ curl $(minikube service --url aladdin-demo)/app/elasticsearch
+    
+    Data from ElasticSearch is {"author": "Aladdin", "song": "A Whole New World", "lyrics": ["I can show you the world"], "awesomeness": 42} 
+    
+Now set `elasticsearch.populate` to `false` and redeploy the app with 
```

Comment:
I like having this elasticsearch.populate toggle-able for the purpose of the tutorial being explicit about what's going on, but maybe we should add that people should make their population scripts idempotent. For example, if I deploy to KDEV, I shouldn't need to have to change some helm variable the next time I deploy if that makes sense.

Reply:
Good point, I added a short blurb at the bottom of this readme.